ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -47,12 +48,17 @@ pub async fn quicksort_by_key_insecure<C, K, F, S>(
 where
     C: Context,
     S: Send + Sync,
-    F: Fn(&S) -> &AdditiveShare<K> + Send + Sync,
+    F: Fn(&S) -> &AdditiveShare<K> + Sync,
     for<'a> &'a AdditiveShare<K>: IntoIterator<Item = AdditiveShare<K::Element>>,
     K: WeakSharedValue + Field + CustomArray<Element = Boolean>,
 {
+    // expected amount of recursion: compute ceil(log_2(list.len()))
+    let expected_depth = std::mem::size_of::<usize>() * 8 - list.len().leading_zeros() as usize;
     // create stack
-    let mut stack: Vec<(C, usize, usize)> = vec![];
+    let mut stack: Vec<(C, usize, usize)> = Vec::with_capacity(expected_depth);
+    // vector for comparison (against pivot) outcomes
+    #[allow(unused_assignments)]
```

Comment:
Rustc is right - you are still reallocating those bit vectors on each iteration. There is no `try_collect(container)`, so you would probably do it by hand

```rust
comp.clear();
while let Some(bool_value) = seq_join(ctx.active_work(), stream_iter(...)).next().await {
    comp.push(bool_value);
}
```